nixos/modules/services/x11/window-managers/xmonad.nix

```diff
@@ -48,13 +56,36 @@ in
         type = lib.types.bool;
         description = "Enable xmonad-{contrib,extras} in Xmonad.";
       };
+
+      config = mkOption {
+        default = null;
+        type = with lib.types; nullOr (either path string);
+        description = ''
+          Configuration from which XMonad gets compiled. If no value
+          is specified, the xmonad config from $HOME/.xmonad is taken.
+          If you use xmonad --recompile, $HOME/.xmonad will be taken as
+          the configuration, but on the next restart of display-manager
+          this config will be reapplied.
+        '';
+        example = ''
+          import XMonad
+
+          main = xmonad defaultConfig
```

Comment:
I think it would be better to recommend `launch` over `xmonad`. Since we're generating the configuration with Nix, we should leave out the compiling a haskell file in the home directory and reloading that `xmonad` does.